pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1054,6 +1054,12 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 			klog.Infof(volumeToMount.GenerateMsgDetailed("MapVolume.WaitForAttach succeeded", fmt.Sprintf("DevicePath %q", devicePath)))
 
 		}
+		// For CSI driver, WaitForAttach returns attachment name as devicePath,
+		// not actual devicePath.
+		var isCSI bool
+		if strings.HasPrefix(devicePath, "csi-") {
+			isCSI = true
+		}
 		// A plugin doesn't have attacher also needs to map device to global map path with SetUpDevice()
```

Comment:
I know this is outside your PR but since we are here re-evaluating the meaning of devicePath etc. as it gets passed through various functions, its meaning dependent on CSI vs non-CSI, can we reword or remove this comment. TBH I don't know what it means

Reply:
Improved the comments.